Core.IntegrationTests/GitBackedTestBase.cs

```diff
@@ -96,15 +97,34 @@ protected static string ExecuteGitCommandWithOutput (string argument)
     psi.RedirectStandardOutput = true;
     psi.RedirectStandardError = true;
 
-    using var command = Process.Start(psi);
-    command.WaitForExit();
-    if (command.ExitCode != 0)
+    var process = new Process
+    {
+        StartInfo = psi
+    };
+    
+    var errorBuilder = new StringBuilder();
+    process.ErrorDataReceived += new DataReceivedEventHandler((sender, e) =>
+    {
+      errorBuilder.Append(e.Data);
+    });
+    
+    process.Start();
+
+    process.BeginErrorReadLine();
+    var output = process.StandardOutput.ReadToEnd();
+
+    //waiting for error reading to finish
+    process.WaitForExit();
+    
+    //waiting for process to finish
+    process.WaitForExit();
```

Comment:
Why are we waiting twice? That second `.WaitForExit` should be a no-op. If there was a reason we need it I think something is not working correctly yet.